showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITGdch.java

```diff
@@ -0,0 +1,183 @@
+
+    // modify and save to a temporary folder
+    converted.set(CA_CERT_JSON_KEY, caCertPath.toAbsolutePath().toString());
+    File tempGdchCredentialFile = tempFolder.newFile(TEMP_CREDENTIAL_JSON_FILENAME);
+    try (FileWriter fileWriter = new FileWriter(tempGdchCredentialFile)) {
+      String preparedJson = converted.toPrettyString();
+      fileWriter.write(preparedJson);
+    }
+
+    // use temp location to instantiate credentials
+    credentials = GdchCredentials.fromStream(new FileInputStream(tempGdchCredentialFile));
+  }
+
+  /**
+   * {@link com.google.api.gax.rpc.ClientContext} sets a credentials object many times before
+   * concluding its creation. double check that they end up being the same
+   */
+  @Test
+  public void testClientWithGdchCredential_keepsCredentials() throws IOException {
+    Exception unexpected = getExceptionFromClientCreation();
+    assertNull(unexpected);
+    assertSame(credentials, client.getSettings().getCredentialsProvider().getCredentials());
+  }
+
+  /**
+   * Confirms setting a valid audience is successful.
+   *
+   * The provider's credentials do not get altered when adding audience.
+   * They are recreated and used only inside `ClientContext`
+   * @throws IOException
+   * @throws URISyntaxException
+   */
+  @Test
+  public void testClientWithGdchCredentialWithValidAudience_correct()
+      throws IOException, URISyntaxException {
+    String audience = "valid-audience";
+    settings = settings.toBuilder().setGdchApiAudience(audience).build();
+    Exception unexpected = getExceptionFromClientCreation();
+    assertNull(unexpected);
+    GdchCredentials fromClient =
+            (GdchCredentials) client.getSettings().getCredentialsProvider().getCredentials();
+
+    assertSame(credentials, fromClient);
+  }
+
+  /**
+   * Confirms that the context recreates and contains a different credential when `gdchApiAudience` is specified
+   * @throws IOException
+   * @throws URISyntaxException
+   */
+  @Test
+  public void testClientWithGdchCredentialWithValidAudience_modifiesContextCredentials()
+      throws IOException, URISyntaxException {
+    String audience = "valid-audience";
+    ClientContext context = ClientContext.newBuilder()
+            .setCredentials(credentials)
+            .build();
+    settings = settings.newBuilder(context).build();
+    Exception unexpected = getExceptionFromClientCreation();
+    assertNull(unexpected);
+    GdchCredentials fromClient =
+            (GdchCredentials) client.getSettings().getCredentialsProvider().getCredentials();
+    URI audienceFromClientProvider = fromClient.getApiAudience();
+    assertNotNull(audienceFromClientProvider);
+    assertTrue(audienceFromClientProvider.equals(new URI(audience)));
+    assertTrue(
+            "GDCH credentials with audience should be GdchCredentials",
+            credentials instanceof GdchCredentials);
+    GdchCredentials fromContext = null;
```

Comment:
@blakeli0 I'm looking for a way to access the final credentials that are created with `ClientContext` as in [here](https://github.com/googleapis/sdk-platform-java/blob/ea05922a7010a274db9e38be3940cc54a02a585b/gax-java/gax/src/main/java/com/google/api/gax/rpc/ClientContext.java#L185-L186). 

I believe subclassing `EchoSettings` or `EchoStub[Settings]` to capture the context and hold it so I can assert some aspects of it may be the way, but I wonder if there is a more straightforward way of accessing the call context. Thanks!

Reply:
I don't think we need to verify the credential from `CientContext` directly, verify it from the client is good enough. In addition, I know you probably want to test the scenario that a `StubSetting` can be created from `ClientContext` but as I mentioned in previous comment, I'm not sure that's a valid case even though we currently support it.

Reply:
Right, my purpose was to confirm that a new `Credentials` is created if I set `gdchApiAudience`. Since the `CredentialsProvider` will always return the initial `Credentials` (without audience), 

I was just looking for a way that `Context.create()` is actually using `Credentials` with audience when specified. 

(`ClientContext.create()` logic on recreating `Credentials` explained [here](https://github.com/googleapis/sdk-platform-java/pull/1642/files#r1204754450))

Reply:
As discussed today, it makes more sense to test credentials from `ClientContext` in a unit test since it's more about the internal workings of gax. I added a special test case